pkg/client/optimizely_user_context.go

```diff
@@ -86,6 +95,24 @@ func (o *OptimizelyUserContext) SetAttribute(key string, value interface{}) {
 	o.Attributes[key] = value
 }
 
+// SetQualifiedSegments clears and adds qualified segments for Optimizely user context
+func (o *OptimizelyUserContext) SetQualifiedSegments(qualifiedSegments []string) {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+	if qualifiedSegments == nil {
+		qualifiedSegments = []string{}
+	}
```

Comment:
`qualifiedSegments` are nullable. When null, it indicates non-determined (fetchSegments are not completed yet). Empty segments is valid one (fetchSegments returns an empty list).